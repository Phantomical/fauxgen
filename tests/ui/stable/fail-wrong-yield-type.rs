
#[fauxgen::generator(yield = i32)]
fn gen() {
    r#yield!("test");
}

fn main() {}
