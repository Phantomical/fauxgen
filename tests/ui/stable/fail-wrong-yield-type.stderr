error[E0308]: mismatched types
 --> tests/ui/stable/fail-wrong-yield-type.rs:4:14
  |
2 | #[fauxgen::generator(yield = i32)]
  |                      ----------- arguments to this method are incorrect
3 | fn gen() {
4 |     r#yield!("test");
  |              ^^^^^^ expected `i32`, found `&str`
  |
help: the return type of this call is `&'static str` due to the type of the argument passed
 --> tests/ui/stable/fail-wrong-yield-type.rs:2:1
  |
2 | #[fauxgen::generator(yield = i32)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
3 | fn gen() {
4 |     r#yield!("test");
  |     ----------------
  |     |        |
  |     |        this argument influences the return type of `yield_`
  |     in this macro invocation
note: method defined here
 --> src/detail/token.rs
  |
  |     pub async fn yield_(self: Pin<&Self>, value: Y) -> A {
  |                  ^^^^^^
  = note: this error originates in the macro `yield` (in Nightly builds, run with -Z macro-backtrace for more info)
